### PR TITLE
Equip CIL with a mapping maintaining variable names before/after alpha conversion.

### PR DESCRIPTION
**Pull request by michael-schwarz**
**This is work by our student @faddeenkov which she undertook as part of her BA thesis.**

The idea is to maintain a mapping between the original name a variable had in the C program and the name it has after the alpha-conversion performed by CIL.
This also adds a list of all CIL variables that are temporary variables created by CIL. Until now, there was no reliable way to see if a variable is a temporary CIL variable or not (at least as far as I am aware).

This seems relevant in quite a number of places:
- It is one of the prerequisites for merging the rest of the work for that thesis and https://github.com/goblint/analyzer/pull/112 into Goblint (semantic search)
- It might be interesting to have this available for the HTML output
- This information might be helpful for generating witnesses (CC: @sim642)

### CODE REVIEW

File: src/frontc/cabs2cil.mli [hunk 1]
```diff
@@ -84,3 +86,26 @@ val typeForCombinedArg: ((string, string) Hashtbl.t -> Cil.typ -> Cil.typ) ref
...
+                                         * and refers to a type. Note that
+                                         * the name of the actual type might
+                                         * be different from foo due to alpha
+                                         * conversion *)
+  | EnvLabel of string                  (* The name refers to a label. This
+                                         * is useful for GCC's locally
+                                         * declared labels. The lookup name
+                                         * for this category is "label foo" *)
+
+val myEnv : (string, envdata * location) Hashtbl.t
```
Comment (michael-schwarz) [inline src/frontc/cabs2cil.mli:111]: This would greatly benefit from a comment explaining what exactly it is.

**Comment by sim642** [comment]
Indeed, this would be good information to have for witness generation to make sure invariants only contain actual program variables and not CIL temporaries. Right now I have an ugly name-based heuristic for that:
https://github.com/goblint/analyzer/blob/a32a031712ca481fea1165e29f482ff782f40b27/src/domains/invariantCil.ml.

File: src/frontc/cabs2cil.mli [hunk 2]
```diff
@@ -35,6 +35,8 @@
  *
  *)
 
+ open Cil
```
Comment (michael-schwarz) [inline src/frontc/cabs2cil.mli:38]: Why `open CIL` and not prefix things with `Cil.*` like in the rest of the file?

**Comment by michael-schwarz** [comment]
The CI status here on Github is not updated for some reason, both builds did in fact pass: https://travis-ci.com/github/goblint/cil/builds/190417919

**Pull request merged**